sqlmesh/dbt/loader.py

```diff
@@ -55,19 +59,53 @@ def _load_models(
 
         context = project.context.copy()
 
+        macros_max_mtime: t.Optional[float] = None
+
         for package_name, package in project.packages.items():
             context.jinja_macros.add_macros(
-                package.macros,
+                package.macro_infos,
                 package=package_name if package_name != context.project_name else None,
             )
+            macro_files_mtime = [
+                self._path_mtimes[m.path]
+                for m in package.macros.values()
+                if m.path in self._path_mtimes
+            ]
```

Comment:
Why not move out of the loop and concat new times and do one max afterwards? Then you don't need the if/else for max.